### PR TITLE
AME-3041 Add support for cluster autoscaler settings

### PR DESCRIPTION
**Pull request by HugUlf**

### CODE REVIEW

File: pkg/acloudapi/apitypes.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ type Cluster struct {
 	ObservabilityTenant          *ObservabilityTenant   `json:"observabilityTenant,omitempty" yaml:"ObservabilityTenant,omitempty"`
 	EnvironmentPrometheusRules   bool                   `json:"environmentPrometheusRules" yaml:"EnvironmentPrometheusRules"`
 	MaintenanceSchedule          *MaintenanceSchedule   `json:"maintenanceSchedule,omitempty" yaml:"MaintenanceSchedule,omitempty"`
+	AutoScalerSettings           AutoscalingSettings    `json:"clusterAutoscalerSettings,omitempty" yaml:"ClusterAutoScalerSettings,omitempty"`
```
Comment (djmarkoz) [inline pkg/acloudapi/apitypes.go]: This should be a pointer 
```suggestion
	AutoScalerSettings           *AutoscalingSettings    `json:"clusterAutoscalerSettings,omitempty" yaml:"ClusterAutoScalerSettings,omitempty"`
```

File: pkg/acloudapi/apitypes.go [hunk 2]
```diff
@@ -152,6 +153,8 @@ type CreateCluster struct {
 	NodePools   []NodePools         `json:"nodePools" yaml:"NodePools"`
 	IPWhitelist []IPWhitelistEntry  `json:"ipWhitelist,omitempty" yaml:"IpWhitelist,omitempty"`
 	Addons      map[string]APIAddon `json:"addons,omitempty" yaml:"Addons,omitempty"`
+
+	AutoScalerSettings AutoscalingSettings `json:"clusterAutoscalerSettings,omitempty" yaml:"ClusterAutoScalerSettings,omitempty"`
```
Comment (djmarkoz) [inline pkg/acloudapi/apitypes.go]: This should be a pointer 

```suggestion
	AutoScalerSettings *AutoscalingSettings `json:"clusterAutoscalerSettings,omitempty" yaml:"ClusterAutoScalerSettings,omitempty"`
```

**Review by djmarkoz — CHANGES_REQUESTED** [review]

File: pkg/acloudapi/apitypes.go [hunk 3]
```diff
@@ -178,6 +180,7 @@ type UpdateCluster struct {
 	DeleteProtection            *bool               `json:"deleteProtection,omitempty" yaml:"DeleteProtection,omitempty"`
 	IPWhitelist                 []string            `json:"ipWhitelist,omitempty" yaml:"IpWhitelist,omitempty"`
 	Addons                      map[string]APIAddon `json:"addons,omitempty" yaml:"Addons,omitempty"`
+	AutoScalerSettings          AutoscalingSettings `json:"clusterAutoscalerSettings,omitempty" yaml:"ClusterAutoScalerSettings,omitempty"`
```
Comment (djmarkoz) [inline pkg/acloudapi/apitypes.go]: I think we should use a pointer here as well.

**Review by idebeijer — APPROVED** [review]
LGTM

**Review by djmarkoz — APPROVED** [review]

**Pull request merged**